Assets/Scripts/TrainingArena.cs

```diff
@@ -99,20 +133,31 @@ public void ResetArena()
 
 		CleanUpSpawnedObjects();
 
-		DetermineNextArenaID();
+		SetNextArenaID();
 
-		if (!TryLoadArenaConfiguration(out ArenaConfiguration newConfiguration))
+		UpdateActiveArenaToCurrentArenaID();
+	}
+
+	public void LoadNextArena()
+	{
+		// TrainingArena must have reset() called at first to initialise arenaID
+		if (isFirstArenaReset)
 		{
-			Debug.LogError("Failed to load arena configuration");
-			return;
+			throw new InvalidOperationException("LoadNextArena called before first reset");
 		}
 
-		ApplyNewArenaConfiguration(newConfiguration);
-
-		CleanupRewards();
+		Debug.Log($"Loading next arena. Previous: {arenaID}, next: {arenaID + 1}");
+		CleanUpSpawnedObjects();
 
-		NotifyArenaChange();
+		arenaID += 1;
+		// We need an arena to sequentially follow the current one to LoadNextArena
+		if (!_environmentManager.GetConfiguration(arenaID, out _))
+		{
+			// TODO: This is the error hit if mergeNextArena is put in the final arena. Add some validation to move this from a runtime error
+			throw new InvalidOperationException($"Tried to LoadNextArena but arena {arenaID} did not exist");
```

Comment:
i think this is a very generic exception type but for your TODO, it might be worthwhile investigating other, more specific and relatable exeptions types. I could also help out with this.

Furthermore, here is what i would put here: 

```suggestion
			throw new KeyNotFoundException($"Tried to LoadNextArena but arena {arenaID} did not exist");
```
This works a bit better as arenaID and the configuration is associated. 
We can even look for custom exceptions which are more informative and relevant. i.e.


```suggestion
    public ArenaConfigurationException(int arenaId)
        : base($"Failed to load configuration for arena {arenaId}.")
    {
        ArenaId = arenaId;
    }
```

Reply:
Thanks, this makes sense - I think probably a custom exception would be a bit overkill to use here, but `KeyNotFoundException` sounds sensible.

Reply:
I think a custom exemption would be useful because we rely so much on this specific logic to work so the arena is  actually instantiated and loaded correctly with the associated arenaID key. having a custom exception class for arena configurations would help and make the code more robust, in general and not just for this script. 

 But this is for another time :)